### PR TITLE
[#2547] Remove escape key close function in the first-run dialog

### PR DESCRIPTION
**Pull request by cadecairos**
Lighthouse: https://webmademovies.lighthouseapp.com/projects/65733/tickets/2547-crash-httpdevbutyrorgsrcbutterjs9872-typeerror-_modal-is-null

### CODE REVIEW

File: src/dialog/dialogs/first-run.js [hunk 1]
```diff
@@ -2,71 +2,11 @@
  *  * If a copy of the MIT license was not distributed with this file, you can
  *  * obtain one at http://www.mozillapopcorn.org/butter-license.txt */
```
Comment (secretrobotron) [inline src/dialog/dialogs/first-run.js]: Mind cleaning this up?

File: src/dialog/dialogs/first-run.js [hunk 2]
```diff
@@ -2,71 +2,11 @@
...
 
-define( [ "text!dialog/dialogs/first-run.html", "dialog/dialog",
-          "ui/widget/tooltip", "util/shims" ],
-  function( LAYOUT_SRC, Dialog, ToolTip ) {
+define( [ "text!dialog/dialogs/first-run.html", "dialog/dialog", "util/shims" ],
+  function( LAYOUT_SRC, Dialog ) {
```
Comment (secretrobotron) [inline src/dialog/dialogs/first-run.js]: And this?

File: src/first-run.js [hunk 3]
```diff
@@ -7,16 +7,91 @@
...
  */
-define( [ "dialog/dialog", "util/cookie" ], function( Dialog, Cookie ) {
+define( [ "dialog/dialog", "util/cookie", "ui/widget/tooltip" ], function( Dialog, Cookie, ToolTip ) {
   return {
     init: function( config ) {
 
-      var _dialog;
+      var TOOLTIP_DELAY = 3000;
+
+      var _dialog,
```
Comment (secretrobotron) [inline src/first-run.js]: Let's just make this `dialog` instead of `_dialog` for consistency's sake.

File: src/dialog/dialogs/first-run.js [hunk 4]
```diff
@@ -2,71 +2,11 @@
  *  * If a copy of the MIT license was not distributed with this file, you can
  *  * obtain one at http://www.mozillapopcorn.org/butter-license.txt */
 
-define( [ "text!dialog/dialogs/first-run.html", "dialog/dialog",
-          "ui/widget/tooltip", "util/shims" ],
-  function( LAYOUT_SRC, Dialog, ToolTip ) {
+define( [ "text!dialog/dialogs/first-run.html", "dialog/dialog", "util/shims" ],
```
Comment (secretrobotron) [inline src/dialog/dialogs/first-run.js]: Does `shims` need to be here anymore?

File: src/first-run.js [hunk 5]
```diff
@@ -7,16 +7,91 @@
...
+            popupTooltip.parentNode.removeChild( popupTooltip );
+          }
+        }, TOOLTIP_DELAY );
+      }
+
+
+      function onDialogClose() {
+        // Remove Listeners
+        dialog.unlisten( "close", onDialogClose );
+        window.removeEventListener( "click", closeDialog, false );
```
Comment (humphd) [inline src/first-run.js:52]: Move this into closeDialog so it happens right after you click.

File: src/first-run.js [hunk 6]
```diff
@@ -7,16 +7,91 @@
...
+        popupTile.removeEventListener( "mouseover", closeDialog, false );
```
Comment (humphd) [inline src/first-run.js:53]: Same here.

File: src/first-run.js (hunk 5 above)
Comment (cadecairos) [inline src/first-run.js:52]: The problem with moving the two lines here is that if they're in closeDialog, when the dialog closes itself via the escape key, closeDialog is never called and the listeners won't be removed

**Pull request merged**